### DISCUSSION TITLE
halo 可以通过idea 打成jar包吗

### DISCUSSION THREAD

**Discussion by ABC15859888** [halo-dev/halo#3534] [Q&A]
halo 怎么才能通过idea 打成jar 包呢

**Comment by ABC15859888** [comment]
这边需要通过jar 的方式去运行halo 博客

**Comment by ruibaby** [comment]
请参考文档：https://docs.halo.run/developer-guide/core/build

**Reply by ABC15859888** [reply]
> 请参考文档：https://docs.halo.run/developer-guide/core/build

多谢， 问题已经得到解决

**Reply by ABC15859888** [reply]
> 请参考文档：https://docs.halo.run/developer-guide/core/build

![image](https://user-images.githubusercontent.com/38815757/226090055-95cbb7e3-8d0d-418a-a5ae-eff02daa5784.png)
   这个是需要17 才能打包吗

**Reply by JohnNiang** [reply]
参考：<https://docs.halo.run/developer-guide/core/prepare>。

**Reply by ABC15859888** [reply]
> 参考：https://docs.halo.run/developer-guide/core/prepare。

这边已经配置了 java 17 jdk  但是依然报错 是需要配置文件指定版本吗 

![image](https://user-images.githubusercontent.com/38815757/226090833-12bc0d07-691a-44b9-8bab-2d213dfb8748.png)

![image](https://user-images.githubusercontent.com/38815757/226090856-54703da4-bda2-4c44-8268-65d4d4ce6362.png)

**Reply by JohnNiang** [reply]
我在 Windows 环境测试了一下，没有任何问题：

```bash
╭─ pwsh  halo   detached at 2c3f9ef ≢  羽3m 58s 148ms⠀                           17.0.4     6   18,16:16 
╰─ﬀ ./gradlew -DsocksProxyHost=localhost -DsocksProxyPort=1080 -q clean build -x check -x jar
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\extension\Unstructured.java:69: warning: This field does not exist, or would have been excluded anyway.
    @EqualsAndHashCode(exclude = "version")
                                 ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
    private final ThemePathPolicy themePathPolicy;
                  ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
    private final ThemePathPolicy themePathPolicy;
                  ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
    private final ThemePathPolicy themePathPolicy;
                  ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
    private final ThemePathPolicy themePathPolicy;
                  ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
    private final ThemePathPolicy themePathPolicy;
                  ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:61: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
        themePathPolicy = new ThemePathPolicy(haloProperties.getWorkDir());
                              ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\extension\router\ExtensionRouterFunctionFactory.java:57: warning: [removal] ListRequest in run.halo.app.extension.router has been deprecated and marked for removal
                    QueryParamBuildUtil.buildParametersFromType(builder, ListRequest.class);
                                                                         ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\security\authorization\DefaultRuleResolver.java:56: warning: [removal] visitRulesFor(UserDetails,RuleAccumulator) in AuthorizationRuleResolver has been deprecated and marked for removal
    public void visitRulesFor(UserDetails user, RuleAccumulator visitor) {
                ^
C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\security\authorization\DefaultRuleResolver.java:41: warning: [removal] rulesFor(UserDetails) in AuthorizationRuleResolver has been deprecated and marked for removal
    public PolicyRuleList rulesFor(UserDetails user) {
                          ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
10 warnings
╭─ pwsh  halo   detached at 2c3f9ef ≢  羽29s 474ms⠀                              17.0.4     9   18,16:18 
╰─ﬀ java -version
java version "17.0.4.1" 2022-08-18 LTS
Java(TM) SE Runtime Environment (build 17.0.4.1+1-LTS-2)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.4.1+1-LTS-2, mixed mode, sharing)
```

**Reply by ABC15859888** [reply]
> 我在 Windows 环境测试了一下，没有任何问题：
> 
> ```shell
> ╭─ pwsh  halo   detached at 2c3f9ef ≢  羽3m 58s 148ms⠀                           17.0.4     6   18,16:16 
> ╰─ﬀ ./gradlew -DsocksProxyHost=localhost -DsocksProxyPort=1080 -q clean build -x check -x jar
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\extension\Unstructured.java:69: warning: This field does not exist, or would have been excluded anyway.
>     @EqualsAndHashCode(exclude = "version")
>                                  ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>     private final ThemePathPolicy themePathPolicy;
>                   ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>     private final ThemePathPolicy themePathPolicy;
>                   ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>     private final ThemePathPolicy themePathPolicy;
>                   ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>     private final ThemePathPolicy themePathPolicy;
>                   ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:49: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>     private final ThemePathPolicy themePathPolicy;
>                   ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\core\extension\reconciler\ThemeReconciler.java:61: warning: [removal] ThemePathPolicy in run.halo.app.theme has been deprecated and marked for removal
>         themePathPolicy = new ThemePathPolicy(haloProperties.getWorkDir());
>                               ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\extension\router\ExtensionRouterFunctionFactory.java:57: warning: [removal] ListRequest in run.halo.app.extension.router has been deprecated and marked for removal
>                     QueryParamBuildUtil.buildParametersFromType(builder, ListRequest.class);
>                                                                          ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\security\authorization\DefaultRuleResolver.java:56: warning: [removal] visitRulesFor(UserDetails,RuleAccumulator) in AuthorizationRuleResolver has been deprecated and marked for removal
>     public void visitRulesFor(UserDetails user, RuleAccumulator visitor) {
>                 ^
> C:\Users\johnn\workspaces\halo-dev\halo\src\main\java\run\halo\app\security\authorization\DefaultRuleResolver.java:41: warning: [removal] rulesFor(UserDetails) in AuthorizationRuleResolver has been deprecated and marked for removal
>     public PolicyRuleList rulesFor(UserDetails user) {
>                           ^
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> Note: Some input files use unchecked or unsafe operations.
> Note: Recompile with -Xlint:unchecked for details.
> 10 warnings
> ╭─ pwsh  halo   detached at 2c3f9ef ≢  羽29s 474ms⠀                              17.0.4     9   18,16:18 
> ╰─ﬀ java -version
> java version "17.0.4.1" 2022-08-18 LTS
> Java(TM) SE Runtime Environment (build 17.0.4.1+1-LTS-2)
> Java HotSpot(TM) 64-Bit Server VM (build 17.0.4.1+1-LTS-2, mixed mode, sharing)
> ```

gradlew -DsocksProxyHost=localhost -DsocksProxyPort=1080 -q clean build -x check -x jar   这个是需要指定地址 和端口吗

**Reply by JohnNiang** [reply]
我只是为了方便下载依赖才配置的 socks5 proxy。你可以不需要这两个配置（socksProxyHost 和 socksProxyPort）。

**Reply by ABC15859888** [reply]
> 我只是为了方便下载依赖才配置的 socks5 proxy。你可以不需要这两个配置（socksProxyHost 和 socksProxyPort）。
主要是打包提示有接口过时导致无法编译  

![image](https://user-images.githubusercontent.com/38815757/226332170-1ab1eb46-789c-40d9-bc3a-d3627ed75807.png)

**Reply by JohnNiang** [reply]
那只是警告而已，可以忽略。不影响正常编译和构建。
